src/lib/checkbox/checkbox.scss

```diff
@@ -240,6 +240,12 @@ $_mat-checkbox-mark-stroke-size: 2 / 15 * $mat-checkbox-size !default;
   ._mat-animation-noopable & {
     transition: none;
   }
+
+  @include cdk-high-contrast {
+    .mat-checkbox.cdk-keyboard-focused & {
```

Comment:
Might be worth mentioning here that this is how IE handles it for the native checkbox.